### PR TITLE
Add a Sonic Adventure file tool

### PR DESCRIPTION
**Pull request by MichaelHinrichs**
Create a page for tools for files from Sonic Adventure in a new folder. Yes, Sonic RPG is the code name for Sonic Adventure. Put SA_Tools on the page.

### CODE REVIEW

**Review by RagdollClash — DISMISSED** [review]
The folder name should be in lowercase.

**Comment by MichaelHinrichs** [comment]
Seriously?

**Comment by TheExentist151** [comment]
Are you sure that SA's codename was sonicRPG?

File: mkdocs.yml [hunk 1]
```diff
@@ -58,6 +58,8 @@ nav:
         - File Formats: tools/retro-engine/common/files.md
         - Models: tools/retro-engine/common/models.md
         - Animation: tools/retro-engine/common/animation.md
+    - Sonic Adventure:
+      - File Formats: tools/sonicRPG/files.md
```
Comment (RagdollClash) [inline mkdocs.yml]:
```suggestion
      - File Formats: tools/sonic_rpg/files.md
```

**Review by RagdollClash — COMMENTED** [review]
Sorry that the capitalization thing came across as rude, I only asked for it to keep things consistent with the rest of the folder names.

That aside, the reference in mkdocs.yml wasn't updated, which is why the action failed.

**Comment by Ahremic** [comment]
> Yes, Sonic RPG is the code name for Sonic Adventure.

Now that's a pretty bold claim. "Sonic and Knuckles RPG" was what the game was called, among a few other things, early in development, but do you have any source that's the game's codename? There doesn't seem to be any trace of that in any strings in SADX's executable, neither the 2004 PC release or the 2010 XBLA version. Word is there's no symbol names for that either. 

Codenames like "Sonic2013" "BlueBlur" "Wars" and "Rangers" etc. all come from verifiable names we can find in the game executable, and even filesystems--Generations' disk folder contains a "bb" folder for "BlueBlur" as an example.

If you have a primary source for this being the _game codename_ that'd be grand.

**Comment by MichaelHinrichs** [comment]
> Are you sure that SA's codename was sonicRPG?

Yes!
https://sonic.fandom.com/wiki/Sonic_Adventure/Beta_elements#Title
https://youtu.be/tlzAEfBBDrI?t=35

**Comment by TheExentist151** [comment]
Wow, didn't know it was a real codename.

**Comment by Ahremic** [comment]
I had a feeling it was going to be a fandom link / youtube video :P

The short answer: Please change the folder name to "sa1" or "sonic" (explained below) instead of "sonic_rpg," or I will have to close this PR, unfortunately.

The long answer:

=============

With respect, this is not the same thing as a project codename. There are no current traces of that project title anywhere else in Adventure's files, executable, or what have you, and that title dates back to the original Saturn version which is very radically different than what the retail game is.

Hedgedocs codenames are based on what we _know_ SEGA have named their games in their _files._ If at some point Sonic Unleashed was called "Sonic Adventure 3" (and according to some reports, it was internally), but that pet-name never made it into the game in any form as a project title, working directory, or anything else, well... that's not exactly a codename we can confirm, is it? For that reason we wouldn't call Unleashed "SA3." Everything about the game is instead "Sonic World Adventure" / "SWA," and SEGA/Sonic Team have been pretty consistent with their project structures... at least the new team has. For that reason Unleashed documentation is all categorized under "SWA."

Generations meanwhile technically has _two_ codenames we are 100% certain about, BlueBlur for the project directory and "Sonic Anniversary" as its beta name--this latter one is still in the files in a leftover debug menu, and is the name of an unearthed scrapped PSP port of the 3DS version. We choose BlueBlur because that's the working directory name we can determine from all their leftover source filepaths etc, but "Sonic Anniversary" would also be an acceptable codename.

As it is, this is more in line with "cute trivia" than anything tangible for what Sonic Adventure *is.* The source on the video you linked meanwhile directs to a (now dead) Sonic Cult page that seems to only source the working title of "SonicRPG" as an anecdote on its "Overview" section, and the Fandom link doesn't even provide a source for that at all. It could be tucked in one of the page's other citations, but regardless.

=============

So all that said, SA1's "codename" is not "SonicRPG" in the same way Forces' codename is "Sonic Wars," or Origins' codename is "HITE" etc. I even looked in the files of both the DC version (pretty empty of any filepath debug info to be honest), both SADX versions, and even SA2PC for some insight.
SADX has references to `f:\DCCNV\SonicAdventure\sonic\src\` on XBLA and `SADXPC\sonic\src` for the 2004 version, and while it's unclear what "DCCNV" could stand for beyond "Dreamcast [something]," this gives the impression SA1's project codename was just "sonic." This is supported by the 2004 exe name being `sonic.exe` and SA2PC having similar, though notably different formatting in its own debug strings. It for example is set up like: `e:\sonicad2\sonicadventure2_gc\sonic2\src`. Note how the PC release's executable name is `sonic2app.exe`, and SADX on Xbox Live Arcade is titled "SonicApp."

Annoyingly I can't find many/any leftover source path strings in the dreamcast files, which would be the smoking gun if we could have that. However, all signs point away from "SonicRPG" being the appropriate name.

I know this may seem kind of hard-assed, but we think it's better for contributions to hedgedocs to be based on hard evidence instead of anecdotal evidence.  That being the case if you can find any source to "SonicRPG" being the working directory of the game, I'd be all for it, but I cannot find that at this time. Otherwise, your contributions are still appreciated!

...And for the record, I probably wouldn't be explaining all this, if your latest commit change wasn't so sassy ;P
Please be respectful or we don't need your help, thanks.

**Comment by thesupersonic16** [comment]
The main point of this PR appears to be about the inclusion of the SA tools, I don't think we need to complain if the name is sonic_rpg unless you have hard proof, personally I think we should take the safe option and use something like `sa1` unless you can provide said verifiable proof right away.

**Comment by MichaelHinrichs** [comment]
This is stupid. Why does this project use codenames anyway? Wouldn't it make sense to use names that people will actually recognize?

**Comment by PTKay** [comment]
Hey! The main reason we prefer using codenames in the HedgeDocs codebase is due to them being, generally, shorter to type when compared to the actual game titles. There's exceptions obviously, and when we cannot find the codename, then we resort to using the game's title.

In this case, since we don't have a codename for this game, I'd recommend that you use `adventure` as the folder name. Be sure to also update the reference to the file in `mkdocs.yml`

**Comment by MichaelHinrichs** [comment]
> I had a feeling it was going to be a fandom link / youtube video :P

Those were the only sources I could find. I looked in better ones first.

**Comment by MichaelHinrichs** [comment]
> Hey! The main reason we prefer using codenames in the HedgeDocs codebase is due to them being, generally, shorter to type when compared to the actual game titles. There's exceptions obviously, and when we cannot find the codename, then we resort to using the game's title.
> 
> In this case, since we don't have a codename for this game, I'd recommend that you use `adventure` as the folder name. Be sure to also update the reference to the file in `mkdocs.yml`

Ok, thank you.

**Pull request merged**